memory.x

```diff
@@ -7,4 +7,8 @@ MEMORY
   /* vectors ( RX )       : ORIGIN = 0x40080000, len = 0x400 */
   iram_seg ( RX )       : ORIGIN = 0x40080400, len = 0xFC00
   dram_seg ( RW )       : ORIGIN = 0x3FFF0000, len = 0x10000
-}
\ No newline at end of file
+
+}
+
+PROVIDE ( rom_i2c_writeReg = 0x400041a4 );
+PROVIDE ( uart_tx_one_char = 0x40009200 );
```

Comment:
This should be mapped directly as a pointer, no?

Reply:
In the mean time I have reimplemented the function in rust